dash-renderer/src/APIController.react.js

```diff
@@ -54,6 +55,7 @@ class UnconnectedContainer extends Component {
                     dispatch
                 );
                 dispatch(setLayout(finalLayout));
+                dispatch(setAppIsReady());
```

Comment:
The layout changes on `SET_LAYOUT`, `UNDO`, `REDO`, `REVERT`. When doing one, do the other.

Reply:
What about on `children` callbacks?

In terms of what kind of layout crawling is required and when it seems to me it matches `applyPersistence` - which also has to crawl the new layout - or layout chunk - in order to find potential matches. We call that immediately above this `setAppIsReady`, but also down in `updateOutput->doUpdateProps`:

https://github.com/plotly/dash/blob/d972c02e8b7842ca9ead4e3965143cac40b5b0b6/dash-renderer/src/actions/index.js#L551-L553

The key for performance there is we're able to crawl only the new piece of layout, and it seems like we should be able to do the same here. If there's any chance the previous components haven't completed loading (perhaps there are 2 simultaneously-invoked children callbacks?) it feels like it should be possible to chain the new promise onto the previous one.

Reply:
https://github.com/plotly/dash/pull/899/commits/b4e105a59bd6e8239377d1c7737f7f7ae73f4e1b